api/repositories/service_binding_repository.go

```diff
@@ -272,6 +280,52 @@ func (r *ServiceBindingRepo) GetServiceBinding(ctx context.Context, authInfo aut
 	return serviceBindingToRecord(*serviceBinding), nil
 }
 
+func (r *ServiceBindingRepo) GetServiceBindingDetails(ctx context.Context, authInfo authorization.Info, bindingGUID string) (ServiceBindingDetailsRecord, error) {
+	userClient, err := r.userClientFactory.BuildClient(authInfo)
+	if err != nil {
+		return ServiceBindingDetailsRecord{}, fmt.Errorf("failed to build user client: %w", err)
+	}
+
+	namespace, err := r.namespaceRetriever.NamespaceFor(ctx, bindingGUID, ServiceBindingResourceType)
+	if err != nil {
+		return ServiceBindingDetailsRecord{}, fmt.Errorf("failed to retrieve namespace: %w", err)
+	}
+
+	serviceBinding := &korifiv1alpha1.CFServiceBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      bindingGUID,
+			Namespace: namespace,
+		},
+	}
+	err = userClient.Get(ctx, client.ObjectKeyFromObject(serviceBinding), serviceBinding)
+	if err != nil {
+		return ServiceBindingDetailsRecord{}, fmt.Errorf("failed to get service binding: %w", apierrors.FromK8sError(err, ServiceBindingResourceType))
+	}
+
+	if ok := isBindingReady(*serviceBinding); !ok {
+		return ServiceBindingDetailsRecord{}, errors.New("binding is not ready yet")
+	}
+
+	credentialsSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      serviceBinding.Status.Credentials.Name,
+		},
+	}
+	err = userClient.Get(ctx, client.ObjectKeyFromObject(credentialsSecret), credentialsSecret)
+	if err != nil {
+		return ServiceBindingDetailsRecord{}, fmt.Errorf("failed to get credentials: %w", apierrors.ForbiddenAsNotFound(apierrors.FromK8sError(err, ServiceBindingResourceType))) //
+	}
+
+	credentials := map[string]any{}
+	err = json.Unmarshal(credentialsSecret.Data[korifiv1alpha1.CredentialsSecretKey], &credentials)
```

Comment:
I would suggest using [this utility](https://github.com/cloudfoundry/korifi/blob/89d5234409cc06abb6462b21cd34cf6f2435212e/tools/secrets.go#L34) to convert secret data into credentials map.